src/compas/geometry/curves/polyline.py

```diff
@@ -578,10 +578,10 @@ def extend(self, length):
         """
         try:
             start, end = length
-            self.points[0] = self.points[0] + self.lines[0].vector.unitized().scaled(-start)
+            self[0] = self[0] + self.lines[0].vector.unitized().scaled(-start)
         except TypeError:
             start = end = length
-        self.points[-1] = self.points[-1] + self.lines[-1].vector.unitized().scaled(end)
+        self[-1] = self[-1] + self.lines[-1].vector.unitized().scaled(end)
```

Comment:
same here. if it is not necessary, i find the code more readable with the explicit reference to points...